src/Compilers/CSharp/Test/Symbol/Compilation/SemanticModelAPITests.cs

```diff
@@ -16,6 +16,87 @@ namespace Microsoft.CodeAnalysis.CSharp.UnitTests
 {
     public class SemanticTests : CSharpTestBase
     {
+        [Fact]
+        public void RefReassignSymbolInfo()
+        {
+            var comp = CreateCompilation(@"
+class C
+{
+    int f = 0; 
+    void M(ref int rx)
+    {
+        (rx = ref f) = 0;
+    }
+}");
+            comp.VerifyDiagnostics();
+            var tree = comp.SyntaxTrees.Single();
+            var root = tree.GetCompilationUnitRoot();
+            var assignment = root.DescendantNodes().OfType<ParenthesizedExpressionSyntax>().Single();
+            var model = comp.GetSemanticModel(tree);
+            Assert.Null(model.GetDeclaredSymbol(assignment));
+            var assignmentInfo = model.GetSymbolInfo(assignment);
+            Assert.Null(assignmentInfo.Symbol);
+        }
+
+        [Fact]
+        public void RefForSymbolInfo()
+        {
+            var comp = CreateCompilation(@"
+class C
+{
+    void M(int x)
+    {
+        for (ref readonly int rx = ref x;;)
+        {
+            if (rx == 0)
+            {
+                return;
+            }
+        }
+    }
+}");
+            comp.VerifyDiagnostics();
+            var tree = comp.SyntaxTrees.Single();
+            var root = tree.GetCompilationUnitRoot();
+            var rx = root.DescendantNodes().OfType<IdentifierNameSyntax>().Last();
```

Comment:
Do we ever also call GetDeclaredSymbol() on the first one and verify symbols are the same?